src/plugins/dashboard/public/application/embeddable/dashboard_container_factory.tsx

```diff
@@ -81,8 +80,11 @@ export class DashboardContainerFactoryDefinition
     initialInput: DashboardContainerInput,
     parent?: Container
   ): Promise<DashboardContainer | ErrorEmbeddable> => {
-    const services = await this.getStartServices();
-    const controlsGroupFactory = services.embeddable.getEmbeddableFactory<
+    const {
```

Comment:
nit: We should be consistent about where the unwrap happens in classes. IMO it should always be in the constructor, even when there is only one usage of `pluginServices.getServices`

Reply:
Addressed in 9631adb454336abb0ac4494bd4ea7b77c8fcf5e8

Reply:
Undid these changes in 86572717582410cc39d7c6c327690add4056d6c2 because the registry isn't actually defined in the dashboard container factory constructor, since this gets defined in the plugin's `start` and not the `setup`. So the registry is `undefined` in the constructor and broke everything hahaha, woopsie 👍